redfish-core/lib/hypervisor_system.hpp

```diff
@@ -213,41 +213,38 @@ inline bool extractHypervisorInterfaceData(
     std::vector<IPv6AddressData>& ipv6Config)
 {
     bool idFound = false;
+
     for (const auto& objpath : dbusData)
     {
-        for (const auto& ifacePair : objpath.second)
-        {
-            IPv4AddressData& ipv4Address = ipv4Config.emplace_back();
-            IPv6AddressData& ipv6Address = ipv6Config.emplace_back();
-
-            for (std::string protocol : {"ipv4", "ipv6"})
+        for (std::string protocol : {"ipv4", "ipv6"})
+        {
+            std::string ipAddrObj = std::format(
+                "/xyz/openbmc_project/network/hypervisor/{}/{}/addr0",
+                ethIfaceId, protocol);
+            IPv4AddressData ipv4Address;
+            IPv6AddressData ipv6Address;
+            if (objpath.first == ipAddrObj)
             {
-                std::string ipAddrObj = std::format(
-                    "/xyz/openbmc_project/network/hypervisor/{}/{}/addr0",
-                    ethIfaceId, protocol);
-                if (objpath.first == ipAddrObj)
+                for (const auto& ifacePair : objpath.second)
```

Comment:
Why is this loop reorder needed?  
The new loop order seems assuming there is only IPv4 & IPv6 address.
Even if it may be true in the current usage, I think  I think the origin loop order seems correct as it can also deal with multiple IP addressses.

```
        for (const auto& ifacePair : objpath.second)
        {
            IPv4AddressData& ipv4Address = ipv4Config.emplace_back();
            IPv6AddressData& ipv6Address = ipv6Config.emplace_back();

            for (std::string protocol : {"ipv4", "ipv6"})
            {
```

I  think what may be needed is the new L276-L289 for origin, and L409-L502 for MacAddress, as the validator is reporting.

Wouldn't this be sufficient?


L276:
```
               if (origin != nullptr && !(origin->empty()))
                            if (protocol == "ipv4")
                                ipv4Address.origin =  translateAddressOriginDbusToRedfish(*origin, true);
                            }
                            if (protocol == "ipv6")
                            {
                                ipv6Address.origin = translateAddressOriginDbusToRedfish(*origin, false);
                            }
```

L502:
 
    if (!ethData.macAddress.empty())
    {
        jsonResponse["MACAddress"] = ethData.macAddress;
    }

Reply:
Done